### PR TITLE
🎉 Source Stripe:: improve 404 handling for `CheckoutSessionsLineItems` stream

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Try to fix this PR https://github.com/airbytehq/oncall/issues/127

There was inconsistency between this 2 HTTP requests:

```
1. https://api.stripe.com/v1/checkout/sessions
2. https://api.stripe.com/v1/checkout/sessions/{checkout_session_id}/line_items
```

First HTTP request returned `session_id=cs_live_HaU2qbxGozr98NLN6PQVtAEP2kJciKAIq337MY763uWCi89kXT08l9lM`
and 2-nd request returned `404` for this session_id. This can happen if during 1-st HTTP  request object existed on server but brefore 2-nd HTTP request object was already removed by somebody else.

## How
Current solution just skip such 404 sub-requests and log them.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-stripe/source_stripe/streams.py [hunk 1]
```diff
@@ -380,7 +380,15 @@ def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs):
         params["expand[]"] = ["data.discounts", "data.taxes"]
         return params
 
+    @property
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-stripe/source_stripe/streams.py:383]: Can we handle the response errors for all streams? It looks like the error could be floating.

**Review by antixar — APPROVED** [review]
LGTM
Please deploy your changes

**Pull request merged**